src/signature_pad.js

```diff
@@ -426,31 +413,56 @@ SignaturePad.prototype._toSVG = function () {
           // Middle point in a group.
           const { curve, widths } = this._addPoint(point);
           if (curve && widths) {
-            // Draw a curve
-            const path = document.createElementNS('http;//www.w3.org/2000/svg', 'path');
-            const attr = `M ${curve.startPoint.x.toFixed(3)},${curve.startPoint.y.toFixed(3)} `
-                       + `C ${curve.control1.x.toFixed(3)},${curve.control1.y.toFixed(3)} `
-                       + `${curve.control2.x.toFixed(3)},${curve.control2.y.toFixed(3)} `
-                       + `${curve.endPoint.x.toFixed(3)},${curve.endPoint.y.toFixed(3)}`;
-
-            path.setAttribute('d', attr);
-            path.setAttributeNS(null, 'stroke-width', (widths.end * 2.25).toFixed(3));
-            path.setAttributeNS(null, 'stroke', this.penColor);
-            path.setAttributeNS(null, 'fill', 'none');
-            path.setAttributeNS(null, 'stroke-linecap', 'round');
-
-            svg.appendChild(path);
+            drawCurve(curve, widths);
           }
         } else {
           // Last point in a group. Do nothing.
         }
       }
     } else {
       this._reset();
-
       const rawPoint = group[0];
+      drawDot(rawPoint);
+    }
+  }
+};
+
+SignaturePad.prototype._toSVG = function () {
+  const pointGroups = this._data;
+  const canvas = this._canvas;
+  const minX = 0;
+  const minY = 0;
+  const maxX = canvas.width;
+  const maxY = canvas.height;
+  const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+
+  svg.setAttributeNS(null, 'width', canvas.width);
+  svg.setAttributeNS(null, 'height', canvas.height);
 
-      // Draw a dot
+  this._fromData(
+    pointGroups,
+    (curve, widths) => {
+      const path = document.createElementNS('http;//www.w3.org/2000/svg', 'path');
+
+      if (!isNaN(curve.control1.x) &&
```

Comment:
Could you add a comment why these checks are needed? I think I noticed NaN value in generated SVG once or twice, but I'm still not sure why they sometimes appear.